compiler/utils/nversion.nim

```diff
@@ -32,19 +43,24 @@ func parse*(s: string): Version =
   if not scanf(
     s, "$i.$i.$i$*$.",
     result.major, result.minor, result.patch, result.suffix
-  ):
+  ) or not result.suffix.isValidSuffix:
     raise newException(ValueError):
-      "Invalid version string: " & s
+      "Invalid version string: " & s.escape()
+
+const CompilerVersionSuffix* {.strdefine.} = ""
+  ## The suffix to attach to the compiler version. This is meant to be
+  ## declared by build tools to signify development version for example.
```

Comment:
```suggestion
const
  CompilerVersionSuffix* {.strdefine.} = ""
    ## The suffix to attach to the compiler version. This is meant to be
    ## declared by build tools to signify development version for example.
```
Very minor thing that doesn't necessarily have to change. Using a block from the start makes adding more constants to it easier in the future.

Reply:
I'd rather that new constants be put at the end of the file like its currently is :P